scripts/release.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+set -e
+
+function get_version {
+  echo $(jq -r .version ./package.json)
+}
+
+function check_tag {
+  local tag=$1
+  published_tags=`curl -s https://api.github.com/repos/$REPO_DOMAIN/$REPO_NAME/tags`
+  already_published=`echo $published_tags | jq ".[] | select(.name == \"$tag\")"`
+  echo $already_published
+}
+
+function release_tag {
+  local tag=$1
+  git fetch --tags
+  local last_tag=`curl -s https://api.github.com/repos/$REPO_DOMAIN/$REPO_NAME/tags | jq --raw-output '.[0].name'`
+  local release_notes=`git log $last_tag..HEAD --oneline`
+  # Parse relase notes like an list (e.g):
+  # - 12345abcd Merge pull request #1 Title
+  #   - 12345efgh commit included 1
+  #   - 12345hijk commit included 2
+  local parsed_release_notes=$(echo "$release_notes" | sed -n -e 'H;${x;s/\n/\\n   - /g;s/^\\n//;p;}')
+  parsed_release_notes=`echo "$parsed_release_notes" | sed -e '${s/  \( - [^ ]* Merge pull request\)/\1/g;}'`
+  release=`curl -H "Authorization: token $ACCESS_TOKEN" -s --data "{
+    \"tag_name\": \"$tag\",
+    \"target_commitish\": \"master\",
+    \"name\": \"$REPO_NAME-$tag\",
+    \"body\": \"Release $tag includes the following commits: \n$parsed_release_notes\",
+    \"draft\": false,
+    \"prerelease\": false
+  }" https://api.github.com/repos/$REPO_DOMAIN/$REPO_NAME/releases`
+  echo $release | jq ".id"
+}
+
+version=`get_version`
+
+if [[ -z "$REPO_NAME" || -z "$REPO_DOMAIN" ]]; then
+  echo "Github repository not specified" > /dev/stderr
+  exit 1
+fi
+
+origin=`git config --get remote.origin.url`
+if [[ "$origin" != "https://github.com/$REPO_DOMAIN/$REPO_NAME" ]]; then
```

Comment:
why is this check necessary?

Reply:
there is also `TRAVIS_REPO_SLUG: The slug (in form: owner_name/repo_name) of the repository currently being built.`

Reply:
I need it to avoid triggering releases from my fork.
Did't find the `TRAVIS_REPO_SLUG`, good point

Reply:
hmm, it shouldn't be running in your fork if you disable builds for it. Also the ACCESS_TOKEN should not be available in yours or anyone else's fork for this to actually release anything if it is enabled. This isn't a check you usually find in travis build files because it generally isn't needed.

Reply:
I use the builds in my fork to early discover errors. I am not sure of how secrets work I know that is not usable in any repository but the one configured but I don't know what  is its value in my fork, that's why I didn't use it for the check. I guess it is not set in that case, I will check that.